STAC-21521 Port Dynatrace integration to API v2

LouisParkin opened the pull request:
### Step 1: Link to Jira issue
https://stackstate.atlassian.net/browse/STAC-21521

### Step 2: Description of changes
Port Dynatrace integration to API v2

### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test	


### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

craffit commented on dynatrace_base/stackstate_checks/dynatrace/dynatrace_client.py:
```diff
@@ -38,6 +38,10 @@ def get_dynatrace_json_response(self, endpoint, params=None):
                 session.verify = self.verify
                 if self.cert:
                     session.cert = (self.cert, self.keyfile)
+                print(f"Getting response from {endpoint}")
```
Not sure this should be in the production code

craffit commented on dynatrace_health/stackstate_checks/dynatrace_health/dynatrace_health.py:
```diff
@@ -9,7 +9,7 @@
 from stackstate_checks.base import StackPackInstance, HealthStream, HealthStreamUrn, Health, Identifiers
 from stackstate_checks.checks import AgentCheck
-from stackstate_checks.dynatrace.dynatrance_client import DynatraceClient
+from stackstate_checks.dynatrace.dynatrace_client import DynatraceClient
```
there is a v1/events call inhere, is that for later?

craffit commented on dynatrace_topology/stackstate_checks/dynatrace_topology/dynatrace_topology.py:
```diff
@@ -8,30 +8,48 @@
+API_V2_DEFAULT_RELATIVE_TIME = '1h'
+API_V2_DEFAULT_FIELDS_STRING = '+fromRelationships,+toRelationships,+tags,+managementZones,+properties'
+
+TOPOLOGY_API_SPEC = {
```
i don't see these changes reflected in the test code, i am expecting the output to have changed somewhat and the mocking needing some work there? I see the tests are failing aswell

LouisParkin committed:
STAC-21521 First attempt at porting Dynatrace Topology to API v2.

STAC-21521 Second attempt at porting Dynatrace Topology to API v2.

STAC-21521 Third attempt at porting Dynatrace Topology to API v2.

STAC-21521 Fourth attempt at porting Dynatrace Topology to API v2.

STAC-21521 Fifth attempt at porting Dynatrace Topology to API v2.

STAC-21521 Sixth attempt at porting Dynatrace Topology to API v2.

STAC-21521 Seventh attempt at porting Dynatrace Topology to API v2.

STAC-21521 Undo passing "self" into validate.

STAC-21521 8th attempt at porting Dynatrace Topology to API v2.

LouisParkin committed:
STAC-21521 Fixed services topology and unit test.

LouisParkin committed:
STAC-21521 Fixed processes topology and unit test.

LouisParkin committed:
STAC-21521 Fixed application topology and unit test.

LouisParkin committed:
STAC-21521 Fixed hosts topology and unit test.

LouisParkin committed:
STAC-21521 Fixed process-groups topology and unit test.

LouisParkin committed:
STAC-21521 Conform custom-device to what was done for it previously.

LouisParkin committed:
STAC-21521 Fix relations test.

LouisParkin committed:
STAC-21521 Fix custom-device test, oversight bug.

LouisParkin committed:
STAC-21521 Slowly fixing paginated custom devices.

LouisParkin committed:
STAC-21521 Fix Custom Device json. Fix application to synthetic monitor relations.

LouisParkin committed:
STAC-21521 Working on e2e topo.

LouisParkin committed:
STAC-21521 Overlooked update to deduplicate hosts.
